lib/api/model/events.dart

```diff
@@ -16,6 +16,7 @@ sealed class Event {
 
   factory Event.fromJson(Map<String, dynamic> json) {
     switch (json['type'] as String) {
+      case 'realm_emoji': return RealmEmojiUpdateEvent.fromJson(json);
```

Comment:
It's true that currently the only possible op for type `realm_emoji` is `update`. But the presence of that `op` field is a way of explicitly leaving space for potentially adding other `op` values for the same type in the future. So let's respect that future-proofing that the API spec has given itself: check for `op` the same as we do for other types, and if it's not the op we expect then call it an unexpected event.

(Fine to skip the intermediate base class, though.)